lib/active_force/active_query.rb

```diff
@@ -1,5 +1,6 @@
 require 'active_support/all'
```

Comment:
Verbal description of code changes:

1. parse the `#select` statement using a new class: `ActiveForce::SelectBuilder.new(selected_fields, self).parse` it returns `nested_query_fields` and `non_nested_query_fields`

2. process `non_nested_query_fields` in the same way we did before. We process `nested_query_fields` by adding it to a new array store on the ActiveQuery object.

3. when a `#includes` is called, we then pass the `nested_query_fields` down into the builders for nested_includes.

4. parse the nested_query_fields to find the specific fields for a specific SObject using the `#query_fields_for(association)`

5. filter out any non selected fields on individual subqueries here for has_many and has_one, and here for belongs_to